### PR TITLE
fixed tf-tests: TestAccRoute53Zone, TestAccRoute53ZoneDataSource

### PR DESCRIPTION
**Pull request by macnev2013**
## TF Tests
  - TestAccRoute53ZoneAssociation
    - TestAccRoute53ZoneAssociation_basic
    - TestAccRoute53ZoneAssociation_disappears
    - TestAccRoute53ZoneAssociation_Disappears_vpc
    - TestAccRoute53ZoneAssociation_Disappears_zone
    - TestAccRoute53ZoneAssociation_crossAccount
    - TestAccRoute53ZoneAssociation_crossRegion
  - TestAccRoute53ZoneDataSource_id
  - TestAccRoute53ZoneDataSource_name
  - TestAccRoute53ZoneDataSource_tags
  - TestAccRoute53ZoneDataSource_vpc

## Fixed
- ListHostedZoneByVPC
- Updated association response to be more precise

### CODE REVIEW

File: moto/route53/models.py [hunk 1]
```diff
@@ -575,10 +575,9 @@ def list_hosted_zones_by_vpc(self, vpc_id):
                 this_zone = self.get_hosted_zone(zone.id)
                 for vpc in this_zone.vpcs:
                     if vpc["vpc_id"] == vpc_id:
-                        this_id = f"/hostedzone/{zone.id}"
                         zone_list.append(
                             {
-                                "HostedZoneId": this_id,
+                                "HostedZoneId": zone.id,
```
Comment (bblommers) [inline moto/route53/models.py:580]: Did you verify this against AWS @macnev2013? [The docs](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/route53.html#Route53.Client.get_hosted_zone) for `get_hosted_zone` specifically say that the output should follow this format, but nothing about the format in `list_hosted_zones_by_vpc`

Comment (macnev2013) [reply moto/route53/models.py:580]: Yes. you can check this doc [here](https://docs.aws.amazon.com/Route53/latest/APIReference/API_ListHostedZonesByVPC.html).
Also verified it with terraform.

Comment (bblommers) [reply moto/route53/models.py:580]: Gotcha. It looks good to me - happy to merge once the unit tests pass

Comment (macnev2013) [reply moto/route53/models.py:580]: Test passed. :)

**Review by bblommers — APPROVED** [review]
Awesome - thanks @macnev2013!

**Pull request merged**